pinot-common/src/main/java/org/apache/pinot/common/utils/PinotDataType.java

```diff
@@ -317,22 +317,24 @@ public Double convert(Object value, PinotDataType sourceType) {
   STRING {
     @Override
     public Integer toInteger(Object value) {
-      return Integer.parseInt((String) value);
+      return Integer.valueOf(((String) value).trim());
```

Comment:
This adds overhead on the happy path (which would be the most common case)? Is it worth enough to only trim in case of exception?

Reply:
Yes. Comparing with the cost of parsing the string, `trim()` should be quite cheap. This is happening on data ingestion instead of query path, so should be fine